src/Fantomas.Core.Tests/CommentTests.fs

```diff
@@ -2446,3 +2446,51 @@ module A =
     type ProviderGeneratedType =
         | ProviderGeneratedType (*ilOrigTyRef*) of ILTypeRef (*ilRenamedTyRef*) * ILTypeRef * ProviderGeneratedType list
 """
+
+[<Test>]
+let ``the one million dollar fix`` () =
+    formatSourceString
+        false
+        """
+[
+    "thing"
+    "thing2"
+    // no other things
+]
+"""
+        { config with MultilineBlockBracketsOnSameColumn = true }
+    |> prepend newline
+    |> should
+        equal
+        """
+[
+    "thing"
+    "thing2"
+    // no other things
+]
+"""
+
+[<Test>]
+let ``this even works nested, oh my`` () =
+    formatSourceString
+        false
+        """
+let v =
+    [
+        "thing"
+        "thing2"
+        // no other things
+    ]
+"""
+        { config with MultilineBlockBracketsOnSameColumn = true }
+    |> prepend newline
+    |> should
+        equal
+        """
+let v =
+    [
+        "thing"
+        "thing2"
+        // no other things
```

Comment:
In order for this to work with newlines before or after the comment, they will probably need to be bundled as one trivia group.